src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java

```diff
@@ -7411,7 +7471,51 @@ public void onError(ServerError error)
    private boolean isWaitingForUserResponseToExternalEdit_ = false;
    private EditingTargetCodeExecution codeExecution_;
    
-   
+   // Timer for autosave
+   private Timer autoSaveTimer_ = new Timer()
+   {
+      @Override
+      public void run()
+      {
+         // It's unlikely, but if we attempt to autosave while running a
+         // previous autosave, just nudge the timer so we try again.
+         if (saving_ != 0)
+         {
+            // If we've been trying to save for more than 5 seconds, we won't
+            // nudge (just fall through and we'll attempt again below)
+            if (System.currentTimeMillis() - saving_ < 5000)
+            {
+               nudgeAutosave();
+               return;
+            }
+         }
+         
+         if (getPath() == null)
+         {
+            // This editor isn't file-backed yet, so there's no save to do.
+            return;
+         }
+         
+         if (docDisplay_.hasActiveCollabSession())
+         {
+            // Everyone's autosave gets turned off during a collab session --
+            // otherwise the autosaves all fire at once and fight
+            return;
+         }
+
+         // Save (and keep track of when we initiated it)
+         saving_ = System.currentTimeMillis();
+         save(() ->
```

Comment:
Similar here: can we handle exceptions during save?

Reply:
👍